### PR TITLE
nGraph some KW fixes

### PR DESCRIPTION
**Pull request by ilyachur**
Cherry-pick of PR: https://github.com/openvinotoolkit/openvino/pull/2176

### CODE REVIEW

**Review by gladilov-gleb — DISMISSED** [review]

File: inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_reshape.cpp [hunk 1]
```diff
@@ -23,6 +23,9 @@ void dynamicToStaticShapeReshape(std::shared_ptr<ngraph::Node> target) {
                      target->get_friendly_name(), target->get_type_info(), ngraph::vpu::op::DynamicShapeResolver::type_info, 0);
 
     const auto reshape = std::dynamic_pointer_cast<ngraph::opset3::Reshape>(target);
+    VPU_THROW_UNLESS(reshape != nullptr,
+        "Cannot cast operation {} of type {} to {}",
+        target->get_friendly_name(), target->get_type_info(), ngraph::opset3::Reshape::type_info);
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_reshape.cpp:28]: https://github.com/openvinotoolkit/openvino/pull/2142#issuecomment-690105396

@Maxim-Doronin please take a look

Comment (Maxim-Doronin) [reply inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_reshape.cpp:28]: Oh, both fixes are similar. 
@ilyachur, may we decide to remove fixes for `vpu/common/src/ngraph` from one of PR's not to conflicts

Comment (ilyachur) [reply inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_reshape.cpp:28]: Ok, cool. I will remove all changes in VPU part

Comment (ilyachur) [reply inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_reshape.cpp]: Removed

File: ngraph/core/src/op/shape_of.cpp [hunk 2]
```diff
@@ -193,6 +193,12 @@ shared_ptr<Node> op::v0::ShapeOf::clone_with_new_inputs(const OutputVector& new_
 {
     check_new_args_count(this, new_args);
     auto new_shape_of = make_shared<op::v0::ShapeOf>(new_args.at(0));
+    NGRAPH_CHECK(new_shape_of.get(),
```
Comment (tomdol) [inline ngraph/core/src/op/shape_of.cpp:196]: What was the error here exactly? I'm just curious why only this op and TensorIterator have this extra check in their clone methods now.

**Review by tomdol — APPROVED** [review]

File: ngraph/core/src/op/shape_of.cpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/core/src/op/shape_of.cpp:196]: KW find issues that potentially `new_shape_of` can be nullptr.

**Review by jane-intel — APPROVED** [review]

**Pull request merged**